ophyd/pseudopos.py

```diff
@@ -327,47 +364,94 @@ def target(self):
         '''Last commanded target positions'''
         return self.PseudoPosition(*(pos.target for pos in self._pseudo))
 
-    def move(self, position, wait=True, timeout=30.0, **kwargs):
-        real_pos = self.forward(position)
-
-        # Remove the 'finished moving' callback, otherwise callbacks will
-        # happen when individual motors finish moving
-        moved_cb = kwargs.pop('moved_cb', None)
-
-        if self.sequential:
-            for real, value in zip(self._real, real_pos):
-                logger.debug('[sequential] Moving %s to %s (timeout=%s)',
-                             real.name, value, timeout)
-                if timeout <= 0:
-                    raise TimeoutError('Failed to move all positioners within '
-                                       '%s s'.format(timeout))
+    def _sequential_move(self, real_pos, timeout=None, **kwargs):
+        '''Move all real positioners to a certain position, in series'''
+        # without proper timeout support for status callbacks, we can't safely
+        # chain motions on other callbacks for sequential motion.  so,
+        # 'sequential_move' above always blocks
+        # this is a big TODO
+
+        self._move_queue[:] = zip(self._real, real_pos)
+        pending_status = []
+        t0 = time.time()
+
+        def move_next(obj=None):
+            # last motion complete message came from 'obj'
+            logger.debug('[%s:sequential] move_next called', self.name)
+            with self._finished_lock:
+                if pending_status:
+                    last_status = pending_status[-1]
+                    if not last_status.success:
+                        logger.error('Failing due to last motion')
+                        self._done_moving(success=False)
+                        return
```

Comment:
As the client you request a motion with `move` and get a status object back. The actual motion process happens asynchronously in the background. If you do a `status.wait()` the wait logic will raise on motion failure. If you don't use that logic, it's up to you to check whether or not the motion succeeded.

Even if an exception were raised here, it would not funnel back where (I think) you're expecting. Well, to be more specific, it would only come back to the caller if the first move were to fail, not any of the subsequent moves.